File: springboot/src/main/java/io/proofmark/showad/cookies/CookieNames.java
package io.proofmark.showad.cookies;

/**
 * Cookie name suffixes used by ShowAd. The full cookie name is built by
 * prefixing each suffix with the configured cookie prefix (default {@code showad}).
 */
public final class CookieNames {

    public static final String FINGERPRINT = "fingerprint";
    public static final String TOKEN = "token";
    public static final String CREATOR = "creator";
    public static final String TICKET = "ticket";
    public static final String VERIFIED = "verified";
    public static final String EXPIRES = "expires";
    public static final String META = "meta";

    private CookieNames() {
    }

    public static String prefixed(String prefix, String suffix) {
        String safePrefix = (prefix == null || prefix.isBlank()) ? "showad" : prefix;
        return safePrefix + "_" + suffix;
    }
}

File: aspnet/src/ShowAd.AspNetCore/Cookies/CookieNames.cs
namespace ShowAd.AspNetCore.Cookies;

public static class CookieNames
{
    public const string Fingerprint = "fingerprint";
    public const string Token = "token";
    public const string Creator = "creator";
    public const string Ticket = "ticket";
    public const string Verified = "verified";
    public const string Expires = "expires";

    public static string Build(string prefix, string suffix) => $"{prefix}_{suffix}";
}
